### ISSUE TITLE
RFE - Add 'filter by intro date of the equipment' option to unit picker

### ISSUE THREAD

**Issue by SimonLandmine** [MegaMek/megamek#2342]
With the change to defining whether units are legal-by-date to being based upon their equipment loadout rather than the 'year introduced' field, would it be possible to add a function (checkbox?) that also allows filtering by unit introduction year (where that is entered on the unit data), so that units that were not introduced by the 'current year' are not available for picking from?
(I know that the intro date is shown on the unit picker, but it just makes it easier to put together 'historically plausible' forces if you're not teased with the unbuilt!)

**Comment by aethernalis** [comment]
Under Advanced Search, you can filter by design year. Is this different/does that not fit the bill?

**Comment by kuronekochomusuke** [comment]
where is the unit introduction year displayed?  not sure which date this is.

**Comment by HammerGS** [comment]
This would be something new and have to be calculated. 

We are currently display the dates like this which is just when the unit as a hole was invented.
![image](https://user-images.githubusercontent.com/13752452/217390162-fe2500ca-b3f3-4ae7-933e-68371f8dcaa6.png)

This RFE would need something different. You'd need to:

- Calculate the earliest possible intro date of every piece of equipment on the unit. 
- Then determine the earliest possible date from those dates that all the equipment could be on the unit.

A wrinkle to this is each piece of equipment has a tech progression if variable tech level is used. The earliest date for Experimental should be earlier than advanced. But if using Static tech levels the earliest date is just the date.

Effectively a unit could be introduced in 3050 but the equipment on it means it could in theory be available in 3040. But under variable tech level it could be Experimental but under Static Tourney Legal. 

But with a quick glance I suspect the Amarok could have actually been around 3101. Really we are showing the earliest the date could exist versus the intro date.

**Comment by HammerGS** [comment]
Would this work?

https://github.com/MegaMek/megameklab/pull/1380

**Comment by SJuliez** [comment]
The calculation would probably work. But we would have to add this somewhat artificial date to our unit cache. It's not a technical problem, but I'm not sure how relevant this really is.

**Comment by SimonLandmine** [comment]
I think there may have been a misunderstanding. I think (vaguely, these several years later) I was asking to be able to filter by 'design year', not 'earliest possible introduction year'. And it appears that 'design year' is the filter option.

(I think it appeared, back in the depths of Oct 2020, that unit date in the picker might have been being based on the equipment loadout, rather than the official design date ... i.e. the reverse to what is actually being done.)

So, in other words, oops. Can probably close this, unless anyone else finds it useful. Sorry!

**Comment by SJuliez** [comment]
I'll just close it then :) can be reopened when necessary
